src/Compilers/CSharp/Portable/Lowering/Instrumentation/DynamicAnalysisInjector.cs

```diff
@@ -129,6 +129,31 @@ private static bool IsExcludedFromCodeCoverage(MethodSymbol method)
             return false;
         }
 
+        private BoundExpression GetFileIndices()
+        {
+            var documents = PooledHashSet<DebugSourceDocument>.GetInstance();
+            foreach (var span in _dynamicAnalysisSpans)
+            {
+                documents.Add(span.Document);
+            }
+
+            if (documents.Count == 0)
+            {
+                // For a compiler generated method that has no 'real' spans, remember the source
+                // document corresponding to the syntax node that is associated with its bound node.
+                documents.Add(GetSourceDocument(_methodBody.Syntax));
+            }
+
+            var fileIndices = ArrayBuilder<BoundExpression>.GetInstance(documents.Count);
+            foreach (var document in documents)
+            {
+                fileIndices.Add(_methodBodyFactory.SourceDocumentIndex(document));
```

Comment:
>fileIndices.Add(_methodBodyFactory.SourceDocumentIndex(document)); [](start = 16, length = 66)

It looks like the order of indices will not be deterministic. Consider sorting content of the fileIndices builder afterwards.